lib/opensearch-aws-sigv4.rb

```diff
@@ -14,45 +14,58 @@
 require 'aws-sigv4/signer'
 require 'faraday'
 require 'json'
+require 'forwardable'
 
 module OpenSearch
   module Aws
     # AWS Sigv4 request signer for <tt>OpenSearch::Transport::Client</tt>.
     #
     # @link https://github.com/opensearch-project/opensearch-ruby/blob/main/DEVELOPER_GUIDE.md#create-a-request-signer
     #
-    # @param [Aws::Sigv4::Signer] sigv4_signer Signer used to sign every request
+    # @param [Hash] options Signer options
+    # @option options [String] :service ('es') The AWS service name
+    # @option options [String] :region ('us-east-1') The AWS region
+    # @option options [String] :access_key_id The AWS access key ID
+    # @option options [String] :secret_access_key The AWS secret access key
+    # @option options [String] :session_token (optional) The AWS session token
     #
     # @example
-    #   signer = Aws::Sigv4::Signer.new(service: 'es',
-    #                                   region: 'us-east-1',
-    #                                   access_key_id: '<access_key_id>',
-    #                                   secret_access_key: '<secret_access_key>',
-    #                                   session_token: '<session_token>')
     #   client = OpenSearch::Client.new({
     #     host: 'https://my-os-domain.us-east-1.es.amazonaws.com/',
-    #     request_signer: OpenSearch::Aws::Sigv4RequestSigner.new(signer)
+    #     request_signer: OpenSearch::Aws::Sigv4RequestSigner.new(
+    #       service: 'es',
+    #       region: 'us-east-1',
+    #       access_key_id: '<access_key_id>',
+    #       secret_access_key: '<secret_access_key>',
+    #       session_token: '<session_token>'
+    #     )
     #   })
     #
     #   puts client.cat.health
     class Sigv4RequestSigner
-      def initialize(sigv4_signer)
-        unless sigv4_signer.is_a?(::Aws::Sigv4::Signer)
-          raise ArgumentError, "Please pass a Aws::Sigv4::Signer. A #{sigv4_signer.class} was given."
-        end
+      extend Forwardable
 
-        @sigv4_signer = sigv4_signer
+      attr_reader :signer
+
+      def_delegators :@signer, :service, :region, :credentials_provider, :unsigned_headers, :apply_checksum_header
+
+      def initialize(options = {})
+        @signer = ::Aws::Sigv4::Signer.new({
+          service: 'es',
+          region: 'us-east-1'
```

Comment:
Have a default service makes sense since this a signer for OpenSearch service. But why a default region, too? From my experience, having an expected default usually leads to confusions for users who forget to set this value and get an error that doesn't properly describe their situation.

Reply:
Makes sense, removed.